### PR TITLE
Week 3 galadoe

### PR DESCRIPTION
**Pull request by galadoe**

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -28,12 +35,34 @@
...
+  let fib = [0, 1]
+  let fibNumber = [1]
+
+  for(let i = 2; i <= num; i++){
+    fib[i] = fib[i - 1] + fib[i - 2];
+    fibNumber.push(fib[i]);
+  }
+  return fibNumber;
+}
+console.log(fibNum(6));
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:54]: Well done!

File: code-challenges.test.js [hunk 2]
```diff
@@ -43,11 +72,57 @@ var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
...
+  let oddNums = []
+  for(let i = 0; i < nums.length; i++){
+    if(typeof nums[i] === "number" && nums[i] % 2 !== 0){
+      oddNums.push(nums[i])
+    }
+  }
+  return oddNums.sort((a,b) => a - b)
+}
+console.log(leastToGreatest(fullArr1));
+console.log(leastToGreatest(fullArr2));
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:90]: Good job!

File: code-challenges.test.js [hunk 3]
```diff
@@ -43,11 +72,57 @@ var fullArr2 = ["hello", 7, 23, -823, false, 78, null, "67", 6, "number"]
...
+//the array is iterated through (or try to use a higher order function to shorten the steps)
+//the previous index is compared to the next index
+//those two index are added and that number is saved and printed (as an array)
+//the saved num is add to the next num in the arrays and printed as well
+//continue to do this until the array is over
+
+const arraySum = (arrayNum) => {
+  let newSum = []
+  for(let i = 1; i < arrayNum.length; i++){
+    if(i === 0){
```
Comment (WalkerTexas-coder) [inline code-challenges.test.js:117]: When I get rid of the conditional get get a few of the values being summed up. You're on the right track but I think this conditional is getting in the way. What did your tests say?

File: interview-questions.md [hunk 4]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is `this` in JavaScript?
 
-  Your answer:
+  Your answer: when we have two+ vars that happen to have the same name, we use 'this' when we are referring to the global variable or the argument
 
-  Researched answer:
+  Researched answer: In most cases, the value of this is determined by how a function is called (runtime binding).
```
Comment (WalkerTexas-coder) [inline interview-questions.md:14]: "this" is something you might want to look back into. It took me a while to get it but "this" refers to the object that the placeholder variable is a part of.  class Person { this.name, this.age, this. height... etc}

File: interview-questions.md [hunk 5]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 2. What is yarn? What file(s) are modified in a React application when you run yarn?
 
-  Your answer:
+  Your answer: yarn allows us to run a react and test file when coding. A node_modules is added when you run yarn
 
-  Researched answer:
+  Researched answer: Yarn is a package manager for your code
```
Comment (WalkerTexas-coder) [inline interview-questions.md:22]: Can you expand on yarn: What other files does it add? How does yarn work?

File: interview-questions.md [hunk 6]
```diff
@@ -9,59 +9,59 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 
 
 3. What are props in React?
 
-  Your answer:
+  Your answer: can be used as an object
 
-  Researched answer:
+  Researched answer: Props are basically kind of global variable or object
```
Comment (WalkerTexas-coder) [inline interview-questions.md:30]: Props are data passed between files in react. Props flow in one direction-- from parent to child component. Every child class component has props built into it where child functional components do not have props unless it is passed in.

File: react-debugging.html [hunk 7]
```diff
@@ -48,13 +48,15 @@
...
         // No need to change anything ABOVE this line ^
+
         render(){
           return(
             <>
               <h1>Draw a Card</h1>
               <Card
                 drawCard={this.drawCard}
                 shuffle={this.shuffle}
+                hand = {this.state.hand}
```
Comment (WalkerTexas-coder) [inline react-debugging.html:59]: Excellent!

File: react-debugging.html [hunk 8]
```diff
@@ -79,10 +81,12 @@ <h1>Draw a Card</h1>
           }
           return(
             <>
-              <button onClick={}>
+            //We were trying to access the props hand when we hadn't created a props through the card component call in the App
+            //also for the onClick, it didn't work because we weren't calling it correctly. It was already being access in the card component in the app, but we weren't calling it as a props in the child component
+              <button onClick={this.props.drawCard}>
```
Comment (WalkerTexas-coder) [inline react-debugging.html:86]: Perfect!

**Review by WalkerTexas-coder — APPROVED** [review]
Hey great job with the assessment! Your code is clean and your tests are great. I can see where your logic is headed and thats excellent. Amazing work on the React debug challenge great pseudo code. Please take your time with the interview questions. The more you write about them the more notes you'll have in the future when you got to prep for interviews. It's super handy and will ease a lot of future stress for you. If you want help elaborating on the topics I recommend asking variations of the big 6 Q's ( Who, What, When, Where, Why, How). Even answering One or two of those will really impress interviewers.  Your PR is approved!

File: interview-questions.md (hunk 5 above)
Comment (galadoe) [reply interview-questions.md:22]: yarn adds dependencies files. When you add yarn, node_modules is added and this contains the files we need for react

File: interview-questions.md (hunk 6 above)
Comment (galadoe) [reply interview-questions.md:30]: without the props, we wouldn't be able to access methods and values from the parent class

**Pull request merged**